pkg/metrics/pushclient/pushclient.go

```diff
@@ -0,0 +1,78 @@
+package pushclient
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+// aggregationClient is a custom HTTP client for use with
+// WeaveWorks Prometheus Aggregation PushGateway
+// We will embed http.Client and essentially
+// "override" the Do function
+type aggregationClient struct {
+	// embed a stdlib http.Client
+	*http.Client
+}
+
+// gatewayURL is the default url that will be used as the endpoint to push to.
+// To push to a centralized prometheus aggregation gateway, change this url.
+var gatewayURL = "http://localhost:9092"
+
+// Do is a custom written client to send data to the prometheus aggregate
+// gateway. It creates a new client and pushes the body that was passed to this function.
+// https://godoc.org/github.com/prometheus/client_golang/prometheus/push#HTTPDoer
+func (a *aggregationClient) Do(req *http.Request) (resp *http.Response, err error) {
+	aggregationGatewayURL := fmt.Sprintf("%s/api/ui/metrics", gatewayURL)
+	newReq, _ := http.NewRequest("PUT", aggregationGatewayURL, req.Body)
+	return a.Client.Do(newReq)
+}
+
+// pushToAggregationGateway is a helper that takes care of the actual code that pushes to the prometheus
+// aggregation gateway. It takes a collector and pushes to the desired url.
+func pushToAggregationGateway(url string, jobName string, collector prometheus.Collector, client push.HTTPDoer) error {
+	err := push.New(url, jobName).
+		Collector(collector).
+		Client(client).
+		Format(expfmt.FmtText).
+		Push()
+	if err != nil {
+		return errors.Wrap(err, "failed to push metrics")
+	}
+	return nil
+}
+
+// pushAllToAggregationGateway is a helper that takes care of the actual code that pushes to the prometheus
+// aggregation gateway. It takes a list of collectors and pushes all of them to the desired url.
+func pushAllToAggregationGateway(url string, collectors []prometheus.Collector, client push.HTTPDoer) error {
```

Comment:
also ` func pushToAggregationGateway(client push.HTTPDoer, url string, jobName string, collectors ...prometheus.Collector) error`
allow you to send one and more collectors using one function.